### PR TITLE
fix(checkbox): Add RequiredTrue validator for md-checkbox

### LINKED ISSUE #4383: template-driven md-checkbox doesn't properly update its validity
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Template-driven `md-checkbox`es should properly handle directive validators placed on them.

#### What is the current behavior?
Currently only reactively defined `md-checkbox`es properly update their validity according to the validator functions on them. Template-driven `md-checkbox`es do not: once they become valid (or if they begin valid), they never update when they should become invalid.

#### What are the steps to reproduce?
See [this plunk](http://plnkr.co/edit/07Uw5ehN6pgXq2MZ2SVy?p=preview).

#### Which versions of Angular, Material, OS, browsers are affected?
all

### PR DESCRIPTION
**Pull request by tinayuangao**
Closes #4383

### CODE REVIEW

File: src/lib/checkbox/checkbox-required-validator.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+@Directive({
+  selector:
+  'md-checkbox[required][formControlName],' +
+  'md-checkbox[required][formControl],md-checkbox[required][ngModel]',
+  providers: [MD_CHECKBOX_REQUIRED_VALIDATOR],
+  host: {'[attr.required]': 'required ? "" : null'}
+})
+export class MdCheckboxRequiredValidator implements Validator {
```
Comment (kara) [inline src/lib/checkbox/checkbox-required-validator.ts]: Have you considered just extending `CheckboxRequiredValidator`? It's exported by the forms module, so you should be able to remove lines 29 - 47, by doing:

```ts
export class MdCheckboxRequiredValidator extends CheckboxRequiredValidator {}
```

Comment (tinayuangao) [inline src/lib/checkbox/checkbox-required-validator.ts]: Considered but AOT doesn't support component inheritance. 
This is the error I got when "extends CheckboxRequiredValidator`: https://travis-ci.org/angular/material2/jobs/228945376

Comment (jelbourn) [inline src/lib/checkbox/checkbox-required-validator.ts]: Hmm, something's not right here. Might be worth asking Tobias for more info

Comment (tinayuangao) [inline src/lib/checkbox/checkbox-required-validator.ts]: Filed issue https://github.com/angular/angular/issues/16637

**Comment by jelbourn** [comment]
@tinayuangao should this PR be closed until the issue in Angular is resolved?

**Comment by tinayuangao** [comment]
Okay. Closed until the issue in Angular is resolved

**Pull request closed without merging**